plugin.rb

```diff
@@ -63,18 +65,36 @@ def vote_count
       end
 
       def user_voted
-        if scope.user
-          object.topic.user_voted(scope.user)
-        else
-          false
-        end
+        scope.user ? object.topic.user_voted?(scope.user) : false
       end
     end
 
+    TopicQuery.results_filter_callbacks << ->(_type, result, user, options) {
+      result = result.includes(:vote_counter)
+      result = result.select("*, (SELECT COUNT(*) AS current_user_voted FROM discourse_voting_votes WHERE user_id = #{user.id} AND topic_id = topics.id)") if user
```

Comment:
I wonder if we need to do a `COUNT(*)` here when there is a unique index on `topic_id` and `user_id` on the `discourse_voting_votes` table. I think checking for existence should be sufficient?

Reply:
I guess it'll be the same as `SELECT 1 AS current_user_voted`... so no big deal here. I would prefer if the value is a boolean instead though so we don't have to do `current_user_voted > 0`